sculture-rest/src/main/java/sculture/controllers/SCultureRest.java

```diff
@@ -342,18 +334,28 @@ public BaseStoryResponse storyVote(@RequestBody StoryVoteRequestBody requestBody
 
         Story ResponseStory = storyDao.voteStory(requestBody.getStory_id(), requestBody.getIsPositive(), requestBody.getUser_id());
 
-        return new BaseStoryResponse(ResponseStory);
+        return new BaseStoryResponse(ResponseStory, tagStoryDao, userDao);
     }
 
     @RequestMapping("/comment/list")
-    public List<CommentResponse> commentList(@RequestBody CommentListRequestBody requestBody) {
-        List<Comment> comments = commentDao.retrieveByStory(requestBody.getStory_id());
-        List<CommentResponse> responses = new LinkedList<CommentResponse>();
+    public CommentListResponse commentList(@RequestBody CommentListRequestBody requestBody) {
+        int page = requestBody.getPage();
+        int size = requestBody.getSize();
+        if (size < 1)
+            size = 10;
+        if (page < 1)
+            page = 1;
+
+        List<Comment> comments = commentDao.retrieveByStory(requestBody.getStory_id(), page, size);
+        List<CommentResponse> responses = new LinkedList<>();
 
-        for (int i = 0; i < comments.size(); i++) {
-            responses.add(new CommentResponse(comments.get(i)));
+        for (Comment comment : comments) {
+            responses.add(new CommentResponse(comment, userDao));
         }
-        return responses;
+
+        CommentListResponse commentListResponse = new CommentListResponse();
+        commentListResponse.setResult(responses);
+        return commentListResponse;
```

Comment:
Previously the response was not JSONObject. It was JSONArray. All other methods returns JSONObject. I have modified that because I don't want to create a special case for this method on Android.